src/object.d

```diff
@@ -3174,6 +3178,28 @@ unittest
    }
 }
 
+/// ditto
+void destroy(T)(ref T* ptr)
+{
+    destroy(*ptr);     // destroy entity referenced by `ptr`
+    ptr = (T*).init;   // destroy `ptr` itself
+}
+
+/// Pointer demonstration
+nothrow @safe unittest
+{
+    static struct S
+    {
+        static int dtorCount;
+        ~this() { dtorCount++; }
+    }
+
+    S* s = new S();
+    destroy(s);
+    assert(s is null);
+    assert(S.dtorCount == 1);
+}
```

Comment:
What I would like to ensure somewhere (doesn't have to be a ddoc unit test), is that the way destroy is implemented, it's not going to recursively follow pointers.

So for instance:
```D
struct S
{
   int *x;
}

S s = new S;
int x = 5;
s.x = &x;
destroy(s);
assert(x == 5);
```

And if I'm reading the code correctly, I think this is going to fail.

Reply:
Just tested this using the implementation in this PR:

```D
void main()
{
    struct S
    {
        int *x;
    }

    S* s = new S;
    int x = 5;
    s.x = &x;
    destroy(s);
    assert(x == 5);
}
```

There is no assertion failure.

Reply:
Added unit test for this scenario

Reply:
Cool, thanks. Yeah, I thought `_destructRecurse` was going to destroy recursively, but it correctly just calls the `__xdtor` of the struct.